src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+const ERROR_MESSAGE = 'Invalid sides for this figure';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
+
+    const sides = [a, b, c];
+    const biggestSide = Math.max(...sides);
+    const sumOfOtherSides = sides.reduce((acc, value) => {
+      if (value !== biggestSide) {
+        return acc + value;
+      }
+
+      return acc;
+    });
+
+    if (biggestSide >= sumOfOtherSides) {
+      throw new Error(ERROR_MESSAGE);
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const triangleSquareFormula = Math.sqrt(p * (p - this.a)
+    * (p - this.b) * (p - this.c));
 
+    return Math.floor((triangleSquareFormula) * 100) / 100;
```

Comment:
Create function for this `Math.floor(...* 100) / 100`. You are doing the same in other places